### PR TITLE
fix(pool): track channels by instance when remote addresses collide

### PR DESCRIPTION
**Pull request by 317787106**
**What does this PR do?**

- Track connection-pool entries by `Channel` instance when different TCP channels have the same remote `InetSocketAddress`.
- Keep active/passive peer counters aligned with the channel that is being connected or disconnected.
- Prevent same-address channels from being collapsed by `Channel.equals()` during pool lifecycle bookkeeping.

**Why are these changes required?**

`Channel.equals()` currently identifies a channel only by its remote socket address. If the remote host supports TCP source-port reuse and uses its listening port as the source port of an inbound connection, an inbound and an outbound channel can have the same remote address while remaining different TCP connections.

The pool could then skip counting the second channel, but later decrement the counter according to the direction of the channel being closed while removing the other channel from the tracked set. Repeated occurrences could leave active/passive tracking inconsistent and affect subsequent connection decisions.

The fix uses channel-instance identity for pool membership and removal. The TCP port-reuse capability is a prerequisite for the address collision; ordinary inbound and outbound connections normally use different remote ports.

**This PR has been tested by:**

- Added lifecycle tests for same-address inbound and outbound channels in both close orders.
- Tested duplicate close notifications, an unregistered channel, replacement notifications, and repeated connect/disconnect cycles.
- Verified that active/passive counters and tracked channel instances remain consistent.
- `git diff --check` passed.

**Follow up**

Keep the existing remote-address mapping in `ChannelManager` under review separately; this PR is limited to connection-pool bookkeeping.

**Extra details**

This change does not modify the wire protocol or require a protocol upgrade.

### CODE REVIEW

**Comment by 3for** [comment]
[SHOULD, java-tron follow-up] Please use channel identity (`==`) in `getPeerConnection()` and identity-based removal throughout `PeerManager`, including `check()`.

Distinct channels sharing a remote address currently resolve to the same `PeerConnection`, which can skip handshake initialization, misassociate messages, and remove another connection's record. Please add tests covering both removal orders and duplicate disconnect callbacks.

**Comment by 317787106** [comment]
> [SHOULD, java-tron follow-up] Please use channel identity (`==`) in `getPeerConnection()` and identity-based removal throughout `PeerManager`, including `check()`.
> 
> Distinct channels sharing a remote address currently resolve to the same `PeerConnection`, which can skip handshake initialization, misassociate messages, and remove another connection's record. Please add tests covering both removal orders and duplicate disconnect callbacks.

I will fix it in java-tron's PR also. If issues, i will comment it here.

**Review by 3for — APPROVED** [review]
LGTM.

File: src/main/java/org/tron/p2p/connection/business/pool/ConnPoolService.java [hunk 1]
```diff
@@ -298,7 +300,10 @@ public void triggerConnect(InetSocketAddress address) {
 
   @Override
   public synchronized void onConnect(Channel peer) {
-    if (!activePeers.contains(peer)) {
+    // cannot use activePeers.contains(peer) to judge whether channel is already exist,
+    // because two channels may have the same hashcode(address + port)
+    // when remote host supporting TCP port reuse
+    if (activePeers.stream().noneMatch(channel -> channel == peer)) {
```
Comment (xxo1shine) [inline src/main/java/org/tron/p2p/connection/business/pool/ConnPoolService.java:304]: Tracking pool entries by channel identity fixes the pool counters, but the same-address case still corrupts ChannelManager's registry: processPeer() replaces the first channel with the second via channels.put(remoteAddress, channel). When the first channel closes, notifyDisconnect() removes by remote address and can delete the still-live second channel. Connection counts and selection then treat that peer as absent. The new tests call ConnPoolService directly, so they do not cover this path. Could ChannelManager's mapping and removal also be made instance-aware, with a test covering both close orders?

Comment (317787106) [reply src/main/java/org/tron/p2p/connection/business/pool/ConnPoolService.java:304]: Fixed in 00a6b85.

  ChannelManager now tracks and removes channels by instance identity. Same-address connections remain separately registered, and closing one no longer removes the other. Connection limits, peer selection, heartbeat iteration, and Status counts now use the complete registry.

  Added regression tests covering both close orders, rejected duplicate handshakes, unregistered-channel closure, and late replacement notifications. The full test suite passed: 80 passed, 4 existing tests skipped.

  `getChannels()` retains its signature but is deprecated and now returns an address-indexed snapshot. Internal callers use `getAllChannels()` or `getChannelCount()`.
